test/runtests.jl

```diff
@@ -635,6 +635,42 @@ end
             @test hash(Box629{Int}(1)) == hash(Box629{Any}(1))
         end
     end
+
+    @testset "other options" begin
+
+        @testset "try user-specified type seed 1" begin
+            @auto_hash_equals typearg=true typeseed=hash struct S640{T}
+                w::T
+            end
+            @test hash(S640{Int}(1)) == hash(1, hash(S640{Int}))
+            @test hash(S640{Int}(1), UInt(2)) == hash(1, UInt(2) + hash(S640{Int}))
+        end
+
+        @testset "try user-specified type seed 2" begin
+            @auto_hash_equals typearg=false typeseed=0x3dfe92e747bd4140 struct S642{T}
+                w::T
+            end
+            @test hash(S642{Int}(1)) == hash(1, 0x3dfe92e747bd4140)
+            @test hash(S642{Int}(1), UInt(2)) == hash(1, UInt(2) + 0x3dfe92e747bd4140)
+        end
+
+        @testset "try user-specified type seed 3" begin
+            @auto_hash_equals typearg=true typeseed=hash struct S650
+                w
+            end
+            @test hash(S650(1)) == hash(1, hash(S650))
+            @test hash(S650(1), UInt(2)) == hash(1, UInt(2) + hash(S650))
+        end
+
+        @testset "try user-specified type seed 2" begin
+            @auto_hash_equals typearg=false typeseed=0x8dfd582f580f2e46 struct S658
```

Comment:
what happens if there's a mismatch between `typearg` and the type of `typeseed` (e.g. false and a function vs true and a scalar) - could we test those as well?